packages/@aws-cdk/aws-cloudformation/test/integ.core-cross-region-references.ts

```diff
@@ -94,3 +95,9 @@ integ.assertions.awsApiCall('CloudFormation', 'deleteStack', {
     ]),
   })).waitForAssertions(),
 );
+
+/**
+ * Test that if we delete more than 10 exports that the
+ * stack will update correctly.
```

Comment:
You may need to add a new integration test (new file and everything) since this test scenario will probably conflict with the one above.

To test this scenario you can probably do a very similar test to the one above, but just invert the scenario. Instead of deleting the producer stack, delete the consumer and then perform a describeStacks and expect the delete to have succeeded.